### PR TITLE
Enable to use custom tracer in FX `symbolic_trace`

### PR DESCRIPTION
**Pull request by regisss**
# What does this PR do?

This PR enables to specify the tracer to use when using `symbolic_trace` and Torch FX. For instance, this can be useful when the user wants a different tracing granularity to not enter some specific modules (e.g. see https://github.com/huggingface/optimum-habana/pull/223). 

## Before submitting
- [ ] This PR fixes a typo or improves the docs (you can dismiss the other checks if that's the case).
- [x] Did you read the [contributor guideline](https://github.com/huggingface/transformers/blob/main/CONTRIBUTING.md#start-contributing-pull-requests),
      Pull Request section?
- [ ] Was this discussed/approved via a Github issue or the [forum](https://discuss.huggingface.co/)? Please add a link
      to it if that's the case.
- [ ] Did you make sure to update the documentation with your changes? Here are the
      [documentation guidelines](https://github.com/huggingface/transformers/tree/main/docs), and
      [here are tips on formatting docstrings](https://github.com/huggingface/transformers/tree/main/docs#writing-source-documentation).
- [ ] Did you write any new necessary tests?

## Who can review?

Anyone in the community is free to review the PR once the tests have passed. Feel free to tag
members/contributors who may be interested in your PR.

### CODE REVIEW

File: src/transformers/utils/fx.py [hunk 1]
```diff
@@ -1207,6 +1207,7 @@ def symbolic_trace(
     model: PreTrainedModel,
     input_names: Optional[List[str]] = None,
     disable_check: bool = False,
+    tracer: HFTracer = None,
```
Comment (michaelbenayoun) [inline src/transformers/utils/fx.py:1210]:
```suggestion
    tracer_cls: Type[HFTracer] = HFTracer,
```

File: src/transformers/utils/fx.py [hunk 2]
```diff
@@ -1240,7 +1243,8 @@ def symbolic_trace(
         check_if_model_is_supported(model)
 
     # Tracing.
-    tracer = HFTracer()
+    if tracer is None:
+        tracer = HFTracer()
```
Comment (michaelbenayoun) [inline src/transformers/utils/fx.py:1247]:
```suggestion
     tracer = tracer_cls()
```

**Review by michaelbenayoun — APPROVED** [review]

**Comment by regisss** [comment]
Pinging @sgugger for final approval.

**Review by sgugger — APPROVED** [review]
Just one formatting nit, thanks for your PR!

File: src/transformers/utils/fx.py [hunk 3]
```diff
@@ -1218,6 +1219,8 @@ def symbolic_trace(
             The names of the inputs of the traced model. If unset, model.dummy_inputs.keys() are used instead.
         disable_check (`bool`, *optional*, defaults to `False`):
             If `True`, no check is done before trying to trace the model, this is mostly usesul for debugging purposes.
+        tracer_cls (Type[HFTracer], *optional*, defaults to `HFTracer`):
```
[the remark this answers is not in the document]
Comment (regisss) [reply src/transformers/utils/fx.py]: Done!

**Pull request merged**